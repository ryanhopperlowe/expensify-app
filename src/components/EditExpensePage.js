import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    console.log('updated', expense);
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemoveExpense = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button onClick={this.onRemoveExpense}>Remove</button>
      </div>
    );
  }
}


const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => (props.match.params.id === expense.id))
});

const mapDispatchToProps = (dispatch) => ({
  editExpense: (expenseId, expense) => dispatch(editExpense(expenseId, expense)),
  removeExpense: ({ id }) => dispatch(removeExpense({ id }))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);