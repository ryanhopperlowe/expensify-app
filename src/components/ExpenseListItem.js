import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3> 
      <Link to={`/edit/${id}`}>{description}</Link>
    </h3>
    <p>{amount} - {createdAt}</p>
  </div>
);

export default connect()(ExpenseListItem);