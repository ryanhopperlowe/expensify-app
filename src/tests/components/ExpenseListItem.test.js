import React from 'react';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';
import { ExpenseListItem } from '../../components/ExpenseListItem';

test('should render expense list item with fixture data', () => {
  const wrapper = shallow(<ExpenseListItem { ...expenses[0] } />);
  expect(wrapper).toMatchSnapshot();
});
