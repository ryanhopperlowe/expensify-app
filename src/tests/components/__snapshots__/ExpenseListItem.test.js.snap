// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item with fixture data 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/1"
    >
      gum
    </Link>
  </h3>
  <p>
    195
     - 
    0
  </p>
</div>
`;
